includes/core-changes/serialization/3.1/jsonserializer-deserialize-throws-jsonexception.md

```diff
@@ -0,0 +1,45 @@
+### JsonSerializer.Deserialize throws JsonException for null or empty strings
+
+When the type parameter is <xref:System.Char>, <xref:System.Text.Json.JsonSerializer.Deserialize%60%601(System.String,System.Text.Json.JsonSerializerOptions)?displayProperty=nameWithType> now throws a <xref:System.Text.Json.JsonException> instead of a <xref:System.NullReferenceException> or <xref:System.IndexOutOfRangeException>.
+
+#### Change description
+
+In previous .NET versions, when the type parameter is <xref:System.Char>, passing a null reference for the <xref:System.String> parameter of <xref:System.Text.Json.JsonSerializer.Deserialize%60%601(System.String,System.Text.Json.JsonSerializerOptions)?displayProperty=nameWithType> causes a <xref:System.NullReferenceException> to be thrown. Passing an empty string causes an <xref:System.IndexOutOfRangeException> to be thrown.
+
+In .NET Core 3.1 and later, when the type parameter is <xref:System.Char>, passing a null reference or an empty string causes a <xref:System.Text.Json.JsonException> to be thrown.
+
+```csharp
+// .NET Core 3.0: Throws NullReferenceException.
+// .NET Core 3.1 and later: Throws JsonException.
+JsonSerializer.Deserialize<char>(null);
```

Comment:
1) This code won't compile without a cast:
> Compilation error (line 10, col 3): The call is ambiguous between the following methods or properties: 'System.Text.Json.JsonSerializer.Deserialize<char>(string, System.Text.Json.JsonSerializerOptions)' and 'System.Text.Json.JsonSerializer.Deserialize<char>(System.ReadOnlySpan<byte>, System.Text.Json.JsonSerializerOptions)'

2) It should be the following:
```suggestion
JsonSerializer.Deserialize<char>("\"null\"");
```

Processing **that** particular JSON string is what was causing a NullReferenceException on 3.0 and is now correctly throwing a JsonException with this message:
> Unhandled exception. System.Text.Json.JsonException: The JSON value could not be converted to System.Char. Path: $ | LineNumber: 0 | BytePositionInLine: 2.

Reply:
Now i confused, isn't `JsonSerializer.Deserialize<char>("\"null\"");` would be be parsed to "null" string instead of `null` value?

I think the original example was the correct one i. e. `JsonSerializer.Deserialize<char>("null");`

Reply:
> I think the original example was the correct one

In this particular case, both are correct and behave the same way (with quotes or without), so you are right, if we want to show an example, either works.

Both were failing with null reference exception in 3.0 and both throw json exception now in 3.1/5.0 (but this change isn't considered breaking). See the example code snippet, below:
https://github.com/dotnet/docs/pull/20956#issuecomment-704669841